lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -0,0 +1,344 @@
+
+  /*
+   * Horrible Hack: When an element within the dropdown is removed from the DOM while it is focused, we want
+   * to move focus to the text input.  It turns out that this is very difficult to track in React, since
+   * useEffect and useLayoutEffect generally fire too late - after the element has already been removed and
+   * lost whatever focus it might've had. The only other way to get this info is with a useLayoutEffect handler
+   * _in the component that was unmounted_, e.g. in NxButton. That would require adding new props for a special use
+   * case to not only NxButton, but also NxLoadWrapper and NxLoadError. Just querying who has focus on every render
+   * seemed like the less bad option.
+   */
+  elFocusedOnMostRecentRender.current = typeof document === 'undefined' ? null : document.activeElement;
+
+  const checkForRemovedFocusedEl = useCallback(function checkForRemovedFocusedEl(mutations: MutationRecord[]) {
+    const nodeContainedFocus = (el: Node) => el.contains(elFocusedOnMostRecentRender.current),
+        nodeListContainedFocus =
+            pipe<[NodeList], Node[], boolean>(Array.from, any(nodeContainedFocus)),
+        focusedChildWasRemoved =
+            any(pipe(prop('removedNodes'), nodeListContainedFocus), mutations);
+
+    if (focusedChildWasRemoved) {
+      focusTextInput();
+    }
+  }, []);
+
+  useMutationObserver(dropdownRef, checkForRemovedFocusedEl, { childList: true });
+
+  function handleOnClick(match: DataItem<T>) {
+    onSelect(match);
+    focusTextInput();
+    setElToFocusId('');
+    setFocusableBtnIndex(null);
+    setShowDropdown(false);
+  }
+
+  function handleElInView(el: HTMLElement) {
+    const bounding = el.getBoundingClientRect();
+    const dropdownMenu = dropdownRef.current;
+
+    //this check if the dropdown menu is not in the document view
+    const isInView = bounding.top >= 0 && bounding.left >= 0 &&
+      bounding.bottom <= (document.documentElement.clientHeight)
+      && bounding.right <= (document.documentElement.clientWidth);
+
+    if (!isInView) {
+      el.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+    }
+
+    //this check if the option is not visible inside the dropdown menu
+    if (dropdownMenu &&
+      dropdownMenu.scrollTop + dropdownMenu.offsetHeight <
+      el.offsetTop + el.offsetHeight
+    ) {
+      el.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+    }
+    else if (dropdownMenu && dropdownMenu.scrollTop > el.offsetTop + 4) {
+      el.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+    }
+  }
+
+  return (
+    /*eslint-disable-next-line jsx-a11y/no-static-element-interactions*/
+    <div ref={ref}
+         className={className}
+         onFocus={handleComponentFocus}
+         onBlur={handleComponentBlur}
+         onMouseDown={handleMouseDown}
+         { ...attrs }>
+      <NxTextInput role="combobox"
+                   ref={inputRef}
+                   id={inputId}
+                   isPristine
```

Comment:
Why is this here?

Reply:
It's removed, and the props are updated here d2bf5beda3a6806fafe467ddf06bd823b966bdcb.